astropy/io/ascii/cds.py

```diff
@@ -126,8 +126,7 @@ def get_cols(self, lines):
                 col.type = self.get_col_type(col)
 
                 match = re.match(
-                    r'\? (?P<equal> =)? (?P<nullval> \S*) (\s+ (?P<descriptiontext> \S.*))?',
-                    col.description, re.VERBOSE)
```

Comment:
Is there a specific reason that you removed this line? Did you run the tests locally before pushing the code to github?

Reply:
As stated in the #9291 issue, i also felt the best way to fix the nullability problem was to change 
r'\? (?P<equal> =)? (?P<nullval> \S*) (\s+ (?P<descriptiontext> \S.*))?', col.description, re.VERBOSE) 
this line to
r'(?:[\[\]] \S* [\[\]])? \? (?P<equal> =)? (?P<nullval> \S*) (\s+ (?P<descriptiontext> \S.*))?'

Reply:
Well, it won't work then. How is this supposed to work? Look up the calling signature of "re.match()" and you'll see that you need to pass in a string that it matches **to**. Please use your own Python programming skills and don't just copy a hint that someone gave in #9291 to copy and paste exactly. That was just that - a hint how to make it work, not a complete programming solution.
And please run the tests locally and make sure they pass (that don't with this!) before you push to github.

Reply:
Okay got it, it won't happen again sorry for the inconvenience caused!